tests/suites/test_suite_pk.data

```diff
@@ -166,55 +166,59 @@ pk_rsa_alt:
 
 Verify ext RSA #12 (PKCS1 v1.5, good)
 depends_on:MBEDTLS_SHA1_C:MBEDTLS_PKCS1_V15
-pk_rsa_verify_ext_test_vec:"206ef4bf396c6087f8229ef196fd35f37ccb8de5efcdb238f20d556668f114257a11fbe038464a67830378e62ae9791453953dac1dbd7921837ba98e84e856eb80ed9487e656d0b20c28c8ba5e35db1abbed83ed1c7720a97701f709e3547a4bfcabca9c89c57ad15c3996577a0ae36d7c7b699035242f37954646c1cd5c08ac":MBEDTLS_MD_SHA1:1024:16:"e28a13548525e5f36dccb24ecb7cc332cc689dfd64012604c9c7816d72a16c3f5fcdc0e86e7c03280b1c69b586ce0cd8aec722cc73a5d3b730310bf7dfebdc77ce5d94bbc369dc18a2f7b07bd505ab0f82224aef09fdc1e5063234255e0b3c40a52e9e8ae60898eb88a766bdd788fe9493d8fd86bcdd2884d5c06216c65469e5":16:"3":"5abc01f5de25b70867ff0c24e222c61f53c88daf42586fddcd56f3c4588f074be3c328056c063388688b6385a8167957c6e5355a510e005b8a851d69c96b36ec6036644078210e5d7d326f96365ee0648882921492bc7b753eb9c26cdbab37555f210df2ca6fec1b25b463d38b81c0dcea202022b04af5da58aa03d77be949b7":MBEDTLS_PK_RSA:-1:MBEDTLS_RSA_SALT_LEN_ANY:0
+pk_rsa_verify_ext_test_vec:"206ef4bf396c6087f8229ef196fd35f37ccb8de5efcdb238f20d556668f114257a11fbe038464a67830378e62ae9791453953dac1dbd7921837ba98e84e856eb80ed9487e656d0b20c28c8ba5e35db1abbed83ed1c7720a97701f709e3547a4bfcabca9c89c57ad15c3996577a0ae36d7c7b699035242f37954646c1cd5c08ac":MBEDTLS_MD_SHA1:1024:16:"e28a13548525e5f36dccb24ecb7cc332cc689dfd64012604c9c7816d72a16c3f5fcdc0e86e7c03280b1c69b586ce0cd8aec722cc73a5d3b730310bf7dfebdc77ce5d94bbc369dc18a2f7b07bd505ab0f82224aef09fdc1e5063234255e0b3c40a52e9e8ae60898eb88a766bdd788fe9493d8fd86bcdd2884d5c06216c65469e5":16:"3":"5abc01f5de25b70867ff0c24e222c61f53c88daf42586fddcd56f3c4588f074be3c328056c063388688b6385a8167957c6e5355a510e005b8a851d69c96b36ec6036644078210e5d7d326f96365ee0648882921492bc7b753eb9c26cdbab37555f210df2ca6fec1b25b463d38b81c0dcea202022b04af5da58aa03d77be949b7":MBEDTLS_PK_RSA:-1:MBEDTLS_RSA_SALT_LEN_ANY:128:0
+
+Verify ext RSA #13 (PKCS1 v2.1, salt_len = max, sig_len too long)
+depends_on:MBEDTLS_PKCS1_V21:MBEDTLS_SHA256_C
+pk_rsa_verify_ext_test_vec:"54657374206d657373616765":MBEDTLS_MD_SHA256:1024:16:"00dd118a9f99bab068ca2aea3b6a6d5997ed4ec954e40deecea07da01eaae80ec2bb1340db8a128e891324a5c5f5fad8f590d7c8cacbc5fe931dafda1223735279461abaa0572b761631b3a8afe7389b088b63993a0a25ee45d21858bab9931aedd4589a631b37fcf714089f856549f359326dd1e0e86dde52ed66b4a90bda4095":16:"010001":"0d2bdb0456a3d651d5bd48a4204493898f72cf1aaddd71387cc058bc3f4c235ea6be4010fd61b28e1fbb275462b53775c04be9022d38b6a2e0387dddba86a3f8554d2858044a59fddbd594753fc056fe33c8daddb85dc70d164690b1182209ff84824e0be10e35c379f2f378bf176a9f7cb94d95e44d90276a298c8810f741c9":MBEDTLS_PK_RSASSA_PSS:MBEDTLS_MD_SHA256:94:129:MBEDTLS_ERR_PK_SIG_LEN_MISMATCH
```

Comment:
Thanks for adding this test. However I don't think it's going to take the PSA path, as it has a requested salt length that's neither `ANY` not the hash length. I think changing `94` to `MBEDTLS_RSA_SALT_LEN_ANY` should do what we want.

Also, I'd like to have a negative test too: if the buffer starts with an invalid signature and we pass a length of 129 instead of 128, the function still returns `MBEDTLS_ERR_RSA_VERIFY_FAILED` (or other) rather than `MBEDTLS_ERR_PK_SIG_LEN_MISMATCH`.

While at it, can we have a test where we pass a smaller length too?

Reply:
You're right, we're taking the old path for this. 
The problem is that for PSA path, `mbedtls_psa_rsa_verify_hash` will not attempt to verify:
```
    if( signature_length != mbedtls_rsa_get_len( rsa ) )
    {
        status = PSA_ERROR_INVALID_SIGNATURE;
        goto exit;
    }
```
Should I restructure it to let it happen, or should we disable this behaviour?

Reply:
To preserve backward compatibility, the caller has to calculate the expected size of the signature (it's very easy for RSA: it's the size of the key), then call the `psa_verify_xxx`. If it returns `PSA_SUCCESS` and the signature is too large, change that to `MBEDTLS_ERR_PK_SIG_LEN_MISMATCH`.

Likewise for ECDSA — except that since https://github.com/ARMmbed/mbedtls/issues/2215 is still not marked as fixed I guess the PSA path doesn't need to implement `PK_SIG_LEN_MISMATCH` for it either.

Reply:
Thanks for your input, @gilles-peskine-arm - I thought we'd like to pass the received `sig_len` no matter what.

Reply:
We need to do that to mimic weird behavior in the legacy interface. Otherwise we would pass the received `sig_len`.

Reply:
Added all requests and more - to handle fallbacks to the original implementation when using max salt.